javascript/javascript3/week3/screenshot.js

```diff
@@ -0,0 +1,197 @@
+class Screenshots {
+    constructor() { 
+        this.urlInput = document.querySelector('.postInput');   
+        this.deleteInput= document.querySelector('.deleteInput');   
+        this.inputs = document.querySelectorAll('input');   
+        this.currentList = document.querySelector('.currentList');  
+        this.fullCollection = document.querySelector('.fullCollection');
+
+        this.screenShotOptions = {
+            method: 'GET',
+            headers: {
+                'X-RapidAPI-Key': 'c4c71f72f4msh1b9d8f48e635255p185dabjsn218387ea6564',
+                'X-RapidAPI-Host': 'website-screenshot6.p.rapidapi.com'
+            }
+        };
+        
+        this.getListOptions = {
+            method: 'GET',	
+        };
+        
+        this.deleteListOptions = {
+            method: 'DELETE',	
+        };
+    }
+
+    renderScreen(id, screenshotUrl, url, user) {
+        const listElement = document.createElement('li');
+        listElement.classList.add('currentListItem');
+        listElement.innerHTML = `
+        ScreenshotUrl - <a href="${screenshotUrl}" target="_blank">${screenshotUrl}</a><br>
+        User URL - <a href="${url}" target="_blank">${url}</a><br>
+        Post id - ${id}<br>
+        User - ${user}
+        `;
+        //------------------------------------
+        this.currentList.append(listElement);
+
+        const img = document.createElement('img');
+        img.src = screenshotUrl;
+        listElement.appendChild(img);
+    }
+
+    renderCollection(id, screenshotUrl, url, user) {
+        const listElement = document.createElement('li');
+        listElement.classList.add('fullCollectionItem');
+        listElement.innerHTML = `id: ${id},<br> screenshotUrl: <a href="${screenshotUrl}" target="_blank">${screenshotUrl}</a>,<br> url: <a href="${url}" target="_blank">${url}</a>,<br> user: ${user}`;
+        this.fullCollection.append(listElement);
+    }
+
+    async getResource(url, options) {
+        let response = await fetch(url, options);
+        if (options.method === 'DELETE') {
+            return null;
+        } else{
+            if (!response.ok) {
+                throw new Error(`Could not fetch ${url}, status: ${response.status}`);
+            }
+        
+            return await response.json();
+        }
+    }
+    
+     async postData(url, data) {
+        let result = await fetch(url, {
```

Comment:
Like above - it is good two wrap every await in a try/catch block. This will be particulary important once you start playing around with node.js as your code runtime environment :) You can rread more about try catch here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch